Userland/Libraries/LibJS/Runtime/PrimitiveString.cpp

```diff
@@ -179,7 +180,7 @@ ThrowCompletionOr<void> PrimitiveString::resolve_rope_if_needed() const
         auto const& rhs_string = TRY(m_rhs->utf16_string());
 
         Utf16Data combined;
-        combined.ensure_capacity(lhs_string.length_in_code_units() + rhs_string.length_in_code_units());
+        TRY_OR_THROW_OOM(vm(), combined.try_ensure_capacity(lhs_string.length_in_code_units() + rhs_string.length_in_code_units()));
```

Comment:
Please `auto& vm = this->vm();` once at the beginning instead of calling it a bunch of times - where the VM comes from is irrelevant when looking at the code that uses it.

Reply:
(This is so that code can consistently refer to `vm` even though we have multiple common sources: function parameter, `realm.vm()`, and `this->vm()`)